### ISSUE TITLE
ERROR: Patches application has failed

### ISSUE THREAD

**Issue by Kerrung** [Frogging-Family/wine-tkg-git#1296]
[prepare.log](https://github.com/user-attachments/files/17508177/prepare.log)
[last_build_config.log](https://github.com/user-attachments/files/17508179/last_build_config.log)
[proton-tkg.log](https://github.com/user-attachments/files/17508181/proton-tkg.log)
![image](https://github.com/user-attachments/assets/9f88bc07-2a6c-49b9-98fe-da1054007263)

**Comment by s0mebodyhelpme** [comment]
Getting a patch application, albeit a different one, too:
`==> WARNING: Skipping OPWR-proton on current childwindow patchset
  -> Applying LAA-unix-staging.patch
  -> Applying proton-tkg-staging-nofsync.patch
==> ERROR: Patch application has failed. The error was logged to /home/al/rep/wine-tkg-git/wine-tkg-git/prepare.log for your convenience.
==> ERROR: A failure occurred in prepare().`

[prepare.log](https://github.com/user-attachments/files/17515041/prepare.log)

**Comment by s0mebodyhelpme** [comment]
Disabling the proton-tkg-staging patchset still yields the following with ntsync patchset:
```
  -> Applying ntsync5-staging.patch
==> ERROR: Patch application has failed. The error was logged to /home/al/rep/wine-tkg-git/wine-tkg-git/prepare.log for your convenience.
==> ERROR: A failure occurred in prepare().
    Aborting...
```
[prepare.log](https://github.com/user-attachments/files/17515060/prepare.log)

**Comment by Tk-Glitch** [comment]
Should be fixed after https://github.com/Frogging-Family/wine-tkg-git/commit/26f451ba853181638f1d49d12ada0ff505aade62 and https://github.com/Frogging-Family/wine-tkg-git/commit/68f0226940abc84696e737f5a53783881ecfd48e

**Comment by Kerrung** [comment]
[proton-tkg.log](https://github.com/user-attachments/files/17528201/proton-tkg.log)
[prepare.log](https://github.com/user-attachments/files/17528205/prepare.log)
[last_build_config.log](https://github.com/user-attachments/files/17528206/last_build_config.log)
68f0226940abc84696e737f5a53783881ecfd48e
![image](https://github.com/user-attachments/assets/ea2aed8d-2689-467e-beab-933b33b4af66)

**Comment by Kerrung** [comment]
@Artewar67 @kangtastic @l12436

**Comment by Tk-Glitch** [comment]
Should be alright with https://github.com/Frogging-Family/wine-tkg-git/pull/1299 merged

**Comment by Kerrung** [comment]
@Tk-Glitch @Artewar67 reopen, please
[last_build_config.log](https://github.com/user-attachments/files/17531767/last_build_config.log)
[prepare.log](https://github.com/user-attachments/files/17531768/prepare.log)
[proton-tkg.log](https://github.com/user-attachments/files/17531769/proton-tkg.log)
34e383deeb978738c8ced6cabc92615159191b93
![image](https://github.com/user-attachments/assets/612078cf-e457-4787-a633-f47b8cd414a6)

**Comment by Kerrung** [comment]
> Should be alright with #1299 merged

As far as I understand, these patches have never been met with the compiler.

**Comment by Tk-Glitch** [comment]
Unable to repro on my end, though there are weirdly missing patches in your log. Please attach your .cfg.

**Comment by Tk-Glitch** [comment]
Built fine here :thinking: 
You may want to try nuking your ccache just in case.
What's your OS and CPU again?

**Comment by Kerrung** [comment]
> Built fine here 🤔 You may want to try nuking your ccache just in case. What's your OS and CPU again?

I use Arch Linux, BTW, and AMD Ryzen 3 4100 × 8

**Comment by Tk-Glitch** [comment]
So that's zen2, while my testing machine is zen3. Maybe worth trying out default compiler flags (-O2 and no -march/-mtune) if clearing ccache doesn't help.
Also if you're using `vulkan-headers-git`, it might be worth trying with Arch's `vulkan-headers` package instead (throwing random ideas out tbh).
If you can capture the actual error when it happens (before the ld exit), that could help narrowing it down.

**Comment by Kerrung** [comment]
> Built fine here 🤔 You may want to try nuking your ccache just in case. What's your OS and CPU again?

I just nuked my ccache and tried again, but the same result, sadly.

**Comment by Kerrung** [comment]
A few days ago I had a compilation running completely without any problems with these settings. I have been using these settings for at least one year. This problem first appeared only today. From this I am making the assumption that there is probably something wrong with these new patches.

> So that's zen2, while my testing machine is zen3. Maybe worth trying out default compiler flags (-O2 and no -march/-mtune) if clearing ccache doesn't help. Also if you're using `vulkan-headers-git`, it might be worth trying with Arch's `vulkan-headers` package instead (throwing random ideas out tbh). If you can capture the actual error when it happens (before the ld exit), that could help narrowing it down.

**Comment by Kerrung** [comment]
I use "4.Wine upstream Proton (Expect breakage)".

**Comment by l12436** [comment]
> @Tk-Glitch @Artewar67 reopen, please [last_build_config.log](https://github.com/user-attachments/files/17531767/last_build_config.log) [prepare.log](https://github.com/user-attachments/files/17531768/prepare.log) [proton-tkg.log](https://github.com/user-attachments/files/17531769/proton-tkg.log) [34e383d](https://github.com/Frogging-Family/wine-tkg-git/commit/34e383deeb978738c8ced6cabc92615159191b93) ![image](https://private-user-images.githubusercontent.com/36989404/380449431-612078cf-e457-4787-a633-f47b8cd414a6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5OTYwNTUsIm5iZiI6MTcyOTk5NTc1NSwicGF0aCI6Ii8zNjk4OTQwNC8zODA0NDk0MzEtNjEyMDc4Y2YtZTQ1Ny00Nzg3LWE2MzMtZjQ3YjhjZDQxNGE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI3VDAyMjIzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyNjAzNTQzMThjMjRkNWYzMjI3MzM4Zjc1MTA0MDhmNGNhZWVlMWQ0YjJkZDE0Y2Q3MWYwZWQ3NzgyNDBlMGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1I2le6-T_RPB32BTsGYBgmO-i5GVBjH9H3Gcr9_b9dM)

save all compile output as a log, the screenshot did not shown the actually error location.

**Comment by Kerrung** [comment]
[file.txt](https://github.com/user-attachments/files/17535158/file.txt)
@l12436 
> save all compile output as a log, the screenshot did not shown the actually error location.

**Comment by Kerrung** [comment]
https://aur.archlinux.org/packages/wine-wow64-git

I've just successfully compiled this vanilla Win package, which also includes winewayland, using exactly the same compiler settings. This means that I have vanilla Wine compiling without any problems. The compilation failure only happens when compiling TKG. I guess I should then disable one patch at a time, among the ones I use when compiling TKG, to find out why. That way the crash should stop after disabling one of the patches, so we can find out which patch is causing the crash.

**Comment by Kerrung** [comment]
After I made _protonify=‘false’ the compilation was became successful. But when it came to the package compression step, it took up all the free space on my drive, where I had 72 GB free, so the package compression could not complete successfully and stopped with a message that there was not enough free space on my drive where the compilation was taking place. I'm still guessing that something is wrong with these patches. There is no way a package can take up more than 72 GB.

[proton-tkg.log](https://github.com/user-attachments/files/17535636/proton-tkg.log)
[last_build_config.log](https://github.com/user-attachments/files/17535637/last_build_config.log)
[prepare.log](https://github.com/user-attachments/files/17535638/prepare.log)

**Comment by l12436** [comment]
error show 
/usr/lib/gcc/x86_64-w64-mingw32/13.1.0/../../../../x86_64-w64-mingw32/bin/ld: dlls/winevulkan/x86_64-windows/loader_thunks.o:loader_thunks.:(.text+0x7c10): multiple definition of `vkGetCalibratedTimestampsEXT'; dlls/winevulkan/x86_64-windows/loader.o:loader.c:(.text+0x16b0): first defined here

seems has patch error cause duplicate function define

**Comment by Kerrung** [comment]
@Tk-Glitch Will this issue be reopened or fixed?

**Comment by Kerrung** [comment]
@Tk-Glitch I just tried it with your patch:

[log2.txt](https://github.com/user-attachments/files/17575513/log2.txt)
[last_build_config.log](https://github.com/user-attachments/files/17575515/last_build_config.log)
[prepare.log](https://github.com/user-attachments/files/17575516/prepare.log)
[proton-tkg.log](https://github.com/user-attachments/files/17575518/proton-tkg.log)

**Comment by Kerrung** [comment]
Maybe my [makepkg.conf](https://github.com/user-attachments/files/17575736/makepkg.conf.zip) can help you?

**Comment by Tk-Glitch** [comment]
Please enable user patches in your .cfg (`_user_patches="false"` -> `_user_patches="true"`) else the patch will not be applied, as shown in your logs.

**Comment by Kerrung** [comment]
> Please enable user patches in your .cfg (`_user_patches="false"` -> `_user_patches="true"`) else the patch will not be applied, as shown in your logs.

Just tried and with your patch this crash have not happend. But all work fine without any third party patches, it means vanilla Wine too: https://github.com/Frogging-Family/wine-tkg-git/issues/1296#issuecomment-2440160933

**Comment by Tk-Glitch** [comment]
> > Please enable user patches in your .cfg (`_user_patches="false"` -> `_user_patches="true"`) else the patch will not be applied, as shown in your logs.
> 
> Just tried and with your patch this crash have not happend. But all work fine without any third party patches, it means vanilla Wine too: [#1296 (comment)](https://github.com/Frogging-Family/wine-tkg-git/issues/1296#issuecomment-2440160933)

Well I am aware and the root cause for *your* issue was found. It's due to the patch above for FH4 which is part of the protonify patchset. Now the weird part is that it builds fine in the CI and on multiple different machines on my end and others - yet not for you. And it's all of a sudden while the last change related to this code was 8 months ago or so on both ends. I don't care about removing that custom vkGetCalibratedTimestampsEXT entry, I care about why it's not an issue for the vast majority of users using that same patchset on the same wine base, and also why it wasn't an issue earlier even though it should be for 8 months.

**Comment by Kerrung** [comment]
> > > Please enable user patches in your .cfg (`_user_patches="false"` -> `_user_patches="true"`) else the patch will not be applied, as shown in your logs.
> > 
> > 
> > Just tried and with your patch this crash have not happend. But all work fine without any third party patches, it means vanilla Wine too: [#1296 (comment)](https://github.com/Frogging-Family/wine-tkg-git/issues/1296#issuecomment-2440160933)
> 
> Well I am aware and the root cause for _your_ issue was found. It's due to the patch above for FH4 which is part of the protonify patchset. Now the weird part is that it builds fine in the CI and on multiple different machines on my end and others - yet not for you. And it's all of a sudden while the last change related to this code was 8 months ago or so on both ends. I don't care about removing that custom vkGetCalibratedTimestampsEXT entry, I care about why it's not an issue for the vast majority of users using that same patchset on the same wine base, and also why it wasn't an issue earlier even though it should be for 8 months.

Please, read it:
https://github.com/Frogging-Family/wine-tkg-git/issues/1296#issuecomment-2439760227
And how I can help you?

**Comment by Kerrung** [comment]
@Tk-Glitch I was able to fix all the problems I described here by deleting the `~/.cache/wine` directory, and mainly these two files
![Screenshot From 2024-11-01 06-05-47](https://github.com/user-attachments/assets/fd3a4365-45d6-4e5b-bc8b-99f498246a5d)
that are created during compilation. But they and this directory had already been created earlier during a previous compilation, apparently with the wrong permissions, which prevented them from being overwritten or accessed for reading. I realised what I should have tried to do after reading what is written here: https://github.com/Frogging-Family/wine-tkg-git/pull/1302 @kangtastic
But if you do what it literally says there, it would completely break ccache for Wine due to the fact that it would only be stored in RAM. Is there any way to add something to the script that would change permissions or delete these files or `~/.cache/wine` directory? I think that would be a better solution than the radical solution of moving the cache directory to RAM. However, since I have everything working properly now, it will be more difficult for me to test it if you do make such changes.
